adminSiteServer/apiRouter.ts

```diff
@@ -2380,12 +2387,17 @@ putRouteWithRWTransaction(apiRouter, "/gdocs/:id", async (req, res, trx) => {
             nextJson.slug
         )
     } else if (checkFullDeployFallback(prevJson, nextJson, hasChanges)) {
```

Comment:
What's a bit weird is that lightning deploys don't trigger a `GdocPublishingAction.Updating` action.
I don't know the numbers, but they might be the more common case than fallback deploys, no?

Reply:
True! I've fixed lightning deploys: a function property was missing from the properties to exclude list and so `isEqual(prev, next)` was always returning `false` - not great that it's so easy to miss this but I think it was mostly due to the large gdocs model refactor and should be easier to notice again now that it's working.

I also extracted functions from the PUT route (probably easier to review in an editor than the git diff)  most notably `indexAndBakeGdocIfNeccesary` which handles all four possible publishing action cases more explicitly, now, and should make it easier to update.